dev_guide/builds.adoc

```diff
@@ -519,9 +544,50 @@ For example, defining a custom HTTP proxy to be used during build:
 ----
 ====
 
+[[source-code]]
+
+== Git Repository Source Options
+
+When the `*BuildConfig.spec.source.type*` is `*Git*`, a Git repository is
+required, and an inline Dockerfile is optional.
+
+The source code is fetched from the location specified and, if the
+`*BuildConfig.spec.source.dockerfile*` field is specified, the inline Dockerfile
+replaces the one in the `*contextDir*` of the Git repository.
+
+The source definition is part of the *spec*` section in the `*BuildConfig*`:
+
+====
+
+----
+{
+  "source" : {
+    "type" : "Git",
+    "git" : { <1>
+      "uri": "git://github.com/openshift/ruby-hello-world.git",
+      "ref": "master"
+    },
+    "contextDir": "app/dir", <2>
+    "dockerfile": "FROM openshift/ruby-22-centos7\nUSER example" <3>
+  },
+}
+----
+
```

Comment:
I just built upon what was already in the document for consistency... if we want to change all examples to another format I'd rather have it done in a separate PR.